pkg/scheduler/framework/plugins/nodeunschedulable/node_unschedulable.go

```diff
@@ -25,8 +25,8 @@ import (
 	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
 )
 
-// NodeUnschedulable is a plugin that priorities nodes according to the node annotation
-// "scheduler.alpha.kubernetes.io/preferAvoidPods".
+// NodeUnschedulable is a plugin that checks if a pod can be scheduled on a
+// unschedulable node according to its tolerations.
```

Comment:
I think "NodeUnschedulable is a plugin that filters pods from being scheduled on nodes with the `node.kubernetes.io/unschedulable` taint." would be a better description as it describes which specific taint would affect this plugin and what the effect is. What do the others think?

Reply:
Yes, according to the code, if the node with the `node.kubernetes.io/unschedulable` taint set to `NoSchedule` and the pod does not tolerate the unschedulable taint, the pod would not be scheduled to the node.

By the way, according to the comment @k82cn 
> TODO (k82cn): deprecates `node.Spec.Unschedulable` in 1.13.
`node.Spec.Unschedulable` would have been deprecated in 1.13. I am wondering if we still need to check node.Spec.Unschedulable here.
```go
   if nodeInfo.Node().Spec.Unschedulable && !podToleratesUnschedulable {
```

Reply:
`node.Spec.Unschedulable` is still used in places like `kubectl cordon`: https://github.com/kubernetes/kubernetes/blob/aa0632208ec0a2191e956d86f3406afbb497b881/staging/src/k8s.io/kubectl/pkg/drain/cordon.go#L83

Reply:
Hi @damemi , thanks for your reply. For this PR, I have updated the comment as suggested by @adtac . Do we need to deprecate `node.Spec.Unschedulable` in another PR ? Maybe I can help with that.

Reply:
> "NodeUnschedulable is a plugin that filters pods from being scheduled on nodes with the node.kubernetes.io/unschedulable taint set to NoSchedule."

This is not accurate, I think it should be: 

"NodeUnschedulable plugin filters nodes that set node.Spec.Unschedulable=true unless the pod tolerates `{key=node.kubernetes.io/unschedulable, effect:NoSchedule}` taint."

Reply:
> Hi @damemi , thanks for your reply. For this PR, I have updated the comment as suggested by @adtac . Do we need to deprecate `node.Spec.Unschedulable` in another PR ? Maybe I can help with that.

I don't know the history, but I think this is why the filter is here, allow the pod to schedule on the pod which is tagged with `node.kubernetes.io/unschedulable` doesn't make sense to me, unless the case @damemi  mentioned, if `node.Spec.Unschedulable` is deprecated we need to figure out alternatives.

Reply:
As mentioned in #69010 , `node.Spec.Unschedulable ` was supposed to be deprecated, but would not be removed until the API v2. To keep backward compatibility [#68984](https://github.com/kubernetes/kubernetes/pull/68984) , we would filter the nodes that set node.Spec.Unschedulable=true unless the pod tolerates {key=node.kubernetes.io/unschedulable, effect:NoSchedule} taint.

Maybe we can fix the comment error here and when `node.Spec.Unschedulable ` is deprecated, we could keep working on that.